.github/workflows/publish-nargo.yml

```diff
@@ -59,21 +59,33 @@ jobs:
       - name: Build environment and Compile
         run: |
           cargo build --package nargo_cli --release --target ${{ matrix.target }} --no-default-features --features "${{ inputs.features }}"
-
+          cargo build --package noir_profiler --release --target ${{ matrix.target }} --no-default-features --features "${{ inputs.features }}"
+      
       - name: Package artifacts
         run: |
           mkdir dist
           cp ./target/${{ matrix.target }}/release/nargo ./dist/nargo
-          7z a -ttar -so -an ./dist/* | 7z a -si ./nargo-${{ matrix.target }}.tar.gz
+          cp ./target/${{ matrix.target }}/release/noir-profiler ./dist/noir-profiler
+
+          # TODO(https://github.com/noir-lang/noir/issues/7445): Remove the separate nargo binary
+          tar -czf nargo-${{ matrix.target }}.tar.gz -C dist nargo
+          tar -czf noir-${{ matrix.target }}.tar.gz -C dist .
 
-      - name: Upload artifact
+      - name: Upload nargo artifact
         uses: actions/upload-artifact@v4
         with:
           name: nargo-${{ matrix.target }}
           path: ./dist/*
           retention-days: 3
 
-      - name: Upload binaries to release tag
+      - name: Upload Noir binaries artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: noir-${{ matrix.target }}
+          path: ./dist/*
+          retention-days: 3
```

Comment:
These artifacts aren't used in the release process tbh. They're just there in case we want to download the binary from this workflow. I think we can just upload `noir-${{ matrix.target }}.tar.gz` in that case.